source/common/http/utility.cc

```diff
@@ -57,8 +57,15 @@ bool Utility::Url::initialize(absl::string_view absolute_url) {
   // RFC allows the absolute-uri to not end in /, but the absolute path form
   // must start with
   if ((u.field_set & (1 << UF_PATH)) == (1 << UF_PATH) && u.field_data[UF_PATH].len > 0) {
```

Comment:
It might be a bit easier to see the differences here if we split out length settings, e.g.

uint64_t path_len = u.field_data[UF_PATH].len;
if ([query param exists) {
  path_len += 1 + u.field_data[UF_PATH].len
}
path_ = path_ = absl::string_view(absolute_url.data() + u.field_data[UF_PATH].off, path_len);

Reply:
yes it looks much better.